Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -1083,13 +1164,150 @@ int Wiimote::CurrentExtension() const
+{
+  bit_value ? value |= (1 << bit_number) : value &= ~(1 << bit_number);
+}
+
+void Wiimote::MotionPlusLogic::Update()
+{
+  if (!IsActive())
+  {
+    return;
+  }
+
+  // TODO: clean up this hackery:
+  // the value seems to increase based on time starting after the first read of 0x00
+  if (IsActive() && times_updated_since_activation < 0xff)
+  {
+    ++times_updated_since_activation;
+
+    // TODO: wtf is this value actually..
+    if (times_updated_since_activation == 9)
+      reg_data.initialization_status = 0x4;
+    else if (times_updated_since_activation == 10)
+      reg_data.initialization_status = 0x8;
+    else if (times_updated_since_activation == 18)
+      reg_data.initialization_status = 0xc;
+    else if (times_updated_since_activation == 53)
+      reg_data.initialization_status = 0xe;
+  }
+
+  auto& mplus_data = *reinterpret_cast<wm_motionplus_data*>(reg_data.controller_data);
+  auto& data = reg_data.controller_data;
+
+  // TODO: make sure a motion plus report is sent first after init
+
+  // On real mplus:
+  // For some reason the first read seems to have garbage data
+  // is_mp_data and extension_connected are set, but the data is junk
+  // it does seem to have some sort of pattern though, byte 5 is always 2
+  // something like: d5, b0, 4e, 6e, fc, 2
+  // When a passthrough mode is set:
+  // the second read is valid mplus data, which then triggers a read from the extension
+  // the third read is finally extension data
+  // If an extension is not attached the data is always mplus data
+  // even when passthrough is enabled
+
+  switch (GetPassthroughMode())
+  {
+  case PassthroughMode::PASSTHROUGH_DISABLED:
+  {
+    mplus_data.is_mp_data = true;
+    break;
+  }
+  case PassthroughMode::PASSTHROUGH_NUNCHUK:
+  {
+    // If we sent mplus data last time now we will try to send ext data.
+    if (mplus_data.is_mp_data)
+    {
+      // The real mplus seems to only ever read 6 bytes from the extension
+      // bytes after 6 seem to be zero filled
+      // The real hardware uses these 6 bytes for the next frame,
+      // but we aren't going to do that
+      if (6 == i2c_bus.BusRead(ACTIVE_DEVICE_ADDR, 0x00, 6, data))
+      {
+        // Passthrough data modifications via wiibrew.org
+        // Data passing through drops the least significant bit of the three accelerometer values
+        // Bit 7 of byte 5 is moved to bit 6 of byte 5, overwriting it
+        SetBit(data[5], 6, Common::ExtractBit(data[5], 7));
+        // Bit 0 of byte 4 is moved to bit 7 of byte 5
+        SetBit(data[5], 7, Common::ExtractBit(data[4], 0));
+        // Bit 3 of byte 5 is moved to  bit 4 of byte 5, overwriting it
+        SetBit(data[5], 4, Common::ExtractBit(data[5], 3));
+        // Bit 1 of byte 5 is moved to bit 3 of byte 5
+        SetBit(data[5], 3, Common::ExtractBit(data[5], 1));
+        // Bit 0 of byte 5 is moved to bit 2 of byte 5, overwriting it
+        SetBit(data[5], 2, Common::ExtractBit(data[5], 0));
+
```

Comment:
is bit 0 of byte 5 set to anything or kept dirty?

Reply:
It's always set to zero below. I'll add a comment.